### PR TITLE
STORE opcode and fancier egl-x11-clear

### PR DESCRIPTION
**Pull request by digetx**
Replace DW hex with a proper asm instruction and make egl-x11-clear (which is I use as a trivial glClear() mesa test) display something other than black.

### CODE REVIEW

File: src/libgrate/grate-font.c [hunk 1]
```diff
@@ -81,11 +81,11 @@ EXEC								\n\
...
 		ipl: t0.fp20, t0.fp20, NOP, NOP			\n\
 								\n\
-	TEX:	tex r2, r3, tex0, r0, r1, r2			\n\
+	TEX:	tex r0, r1, tex0, r0, r1, r2			\n\
 								\n\
-	ALU:	ALU0:	MAD kill, r3.h, #1, #0 (eq)		\n\
+	ALU:	ALU0:	MAD kill, r1.h, #1, #0 (eq)		\n\
 								\n\
-	DW:	0x00028005					\n\
+	DW:	store rt1, r0, r1				\n\
```
Comment (kusma) [inline src/libgrate/grate-font.c:88]: Why not `store rt1, r2, r3` and leave the TEX and ALU opcodes alone?

**Review by kusma — COMMENTED** [review]
Awesome work :)

File: src/libgrate/grate-font.c (hunk 1 above)
Comment (digetx) [reply src/libgrate/grate-font.c:88]: Solely to have a testcase of that, currently we are using r2,r3 elsewhere. It could be an asm_test though.

**Comment by digetx** [comment]
Thank you ;)

File: src/libgrate/grate-font.c (hunk 1 above)
Comment (kusma) [reply src/libgrate/grate-font.c:88]: Ah, that's a good reason :)

**Comment by kusma** [comment]
Hmm, I get an undefined symbol about `sinf@GLIBC_2.4` in egl-x11-clear after this change...  I need to link with `-lm` on my setup...

**Comment by digetx** [comment]
Oh, good catch. I only compiled-tested it with the latest build environment which haven't complained about it, will fix it shortly.

**Comment by digetx** [comment]
Added -lm, compiles fine on 12.04 now ;)

**Pull request merged**